drift/example/main.dart

```diff
@@ -5,18 +5,67 @@ import 'package:drift/native.dart';
 
 part 'main.g.dart';
 
+class TodoCategories extends Table {
+  IntColumn get id => integer().autoIncrement()();
+  TextColumn get name => text()();
+}
+
 class TodoItems extends Table {
   IntColumn get id => integer().autoIncrement()();
   TextColumn get title => text()();
   TextColumn get content => text().nullable()();
+  IntColumn get categoryId => integer().references(TodoCategories, #id)();
+
+  TextColumn get generatedText => text().nullable().generatedAs(
+      title + const Constant(' (') + content + const Constant(')'))();
+}
+
+abstract class TodoCategoryItemCount extends View {
+  TodoItems get todoItems;
+  TodoCategories get todoCategories;
+
+  IntColumn get itemCount => integer().generatedAs(todoItems.id.count())();
+
+  @override
+  Query as() => select([
+        todoCategories.name,
+        itemCount,
+      ]).from(todoCategories).join([
+        innerJoin(todoItems, todoItems.categoryId.equalsExp(todoCategories.id),
+            useColumns: false)
+      ]);
+}
+
+@DriftView(name: 'customViewName')
+abstract class TodoItemWithCategoryNameView extends View {
+  TodoItems get todoItems;
+  TodoCategories get todoCategories;
+
+  TextColumn get title => text().generatedAs(todoItems.title +
+      const Constant('(') +
+      todoCategories.name +
+      const Constant(')'))();
+
+  @override
+  Query as() => select([todoItems.id, title]).from(todoItems).join([
+        innerJoin(
+            todoCategories, todoCategories.id.equalsExp(todoItems.categoryId),
+            useColumns: false)
+      ]);
```

Comment:
I don't fully understand the need for having both column getters and a `as()` method defining the query. Of course, we need to know the overall schema of the view at compile-time, so there's some use for column getters. But to me it feels like giving them a `generatedAs` might conflict with the `Query as()` method.

Maybe there's a way to only define columns and their types with getters and then define all of the view in a single method for consistency?
Maybe something like

```dart
abstract class MyView extends View {
  // define injected tables that we reference
  TableA get a;
  TableB get b;

  // define columns generated by this view
  TextColumn get firstColumn;
  TextColumn get secondColumn;

  // Have a special API for defining the query
  ViewQuery as(DatabaseConnectionUser db) {
    return ViewQuery(
       bindColumns: [firstColumn, secondColumn], // for explicit ordering
       query: db.select(...), // The columns returned by this query are bound to the columns defined by this view
    );
  }
}
```

That way,

- we can just use the unmodified `as` method in the migrator, passing the actual database instance
- we don't need to analyze the `as` method at compile time at all, simplifying the generation process
- we still have all the information necessary to extract the view's schema for compile-time analysis
- IMO, we can make it a runtime error if `bindColumns` does not contain all columns, or if `query` returns the wrong amount of columns.

Sorry for delivering feedback this fundamental so late. I hope an approach like this could actually simplify the view parser and generator in `drift_dev` a bit. Knowing how to define views in SQL, defining expressions for view columns in the select statement feels a bit more intuitive to me.

Reply:
My current implementation use Drift file parser logic and easily integrates to the current Drift logic. The declared views are create DB level views (by CREATE VIEW). The suggested "query only" implementation can't achieve that. Also, this binding solution can cause unexpected behaviours: If the underlying query changes, the view binding can change without the user's knowledge. On the other hand this binding syntax make user lazy. Can query a lot of columns (if use `select()` and not `selectOnly()..addColumns()`) but read only some. Lost all view benefits. 

Migration is not an aspect. On schema change simply drop and recreate all views. 

I can suggest a modify to simplify column declaration:
```dart 
Expression get title => todoItems.title +
      const Constant('(') +
      todoCategories.name +
      const Constant(')'));

   @override
  Query as() => select([todoItems.id, title]).from(todoItems).join([
        innerJoin(
            todoCategories, todoCategories.id.equalsExp(todoItems.categoryId),
            useColumns: false)
      ]);
```

Reply:
> I can suggest a modify to simplify column declaration:

Oh yes, I like that simplified syntax!